minicopier-master/src/minicopier/Configuration.java

```diff
@@ -0,0 +1,187 @@
+	
+	private static final String configName = "configuration.ini";
+	
+	public static String language;
+	
+	public static boolean alwaysOnTop;
+	
+	//Look And Feel
+	//"pgs" (default)
+	//"native"
+	//"metal"
+	public static String lookAndFeel;
+	
+	//What to do if destination file already exists
+	// 0 -> ask user
+	// 1 -> ignore
+	// 2 -> overwrite
+	// 3 -> overwrite if older
+	// 4 -> resume
+	// 5 -> rename
+	public static int collisionAction;
+	
+	//What to do if a source path contains a symbolic link
+	//(Unix systems only)
+	// 0 -> follow
+	// 1 -> ignore
+	public static int symbolicLinkAction;
+    
+    //Is speed limit active?
+    public static boolean limitSpeed;
+    
+    //Maximum throughput of the transfer
+    public static int nbBytesPerS;
+    
+    private static boolean firstLaunch = false;
+	
+	public Configuration(){
+        
+        this.limitSpeed = false;
+        
+        this.nbBytesPerS = 1024*1024;
+		
+		Properties prop = new Properties(); 
+		
+		File configFile = new File(configName);
+		
+		if (!configFile.exists()) {
+			System.out.println("Configuration file not found. Creating configuration.ini.");
+			createConfigFile();
+			firstLaunch = true;
+		}
+		
+		try {
+			FileInputStream in = new FileInputStream(configName);
+			prop.load(in);
+			in.close(); 
+		} catch (Exception e){
+			e.printStackTrace();
+		}
+		
+		setProp(prop);
+		
+	}
+	
+	public static void setProp(Properties prop){
+		
+		String readProp;
+		
+		//Language
+		readProp = prop.getProperty("language");
+		if (("").equals(readProp)||(readProp==null)){
+			language = "SystemLocale";
+		} else {
+			language = readProp;
+		}
+		
+		//Always on top
+		readProp = prop.getProperty("alwaysontop");
+		if (("true").equals(readProp)){
+			alwaysOnTop = true;
+		} else {
+			alwaysOnTop = false;
+		}
+		
+		//Look an Feel
+		readProp = prop.getProperty("lnf");
+		if (("").equals(readProp)||(readProp==null)){
+			lookAndFeel = "pgs";
+		}
+		else {
+			lookAndFeel = readProp;
+		}
+		
+		//Collision action
+		readProp = prop.getProperty("collision");
+		if (("ask").equals(readProp)){
+			collisionAction = 0;
+		} else if (("cancel").equals(readProp)){
+			collisionAction = 1;
+		} else if (("overwrite").equals(readProp)){
+			collisionAction = 2;
+		} else if (("overwrite_older").equals(readProp)){
+			collisionAction = 3;
+		} else if (("resume").equals(readProp)){
+			collisionAction = 4;
+		} else {
+			collisionAction = 0;
+		}
+		
+		//Symbolic links action
+		readProp = prop.getProperty("symlinks");
+		if (("follow").equals(readProp)){
+			symbolicLinkAction = 0;
+		} else if (("ignore").equals(readProp)){
+			symbolicLinkAction = 1;
+		} else {
+			symbolicLinkAction = 0;
+		}
+		
+	}
+	
+	public final void createConfigFile(){
+		InputStreamReader isr=new InputStreamReader(getClass()
+				.getClassLoader().getResourceAsStream("minicopier/configuration.ini"));
+		
+		BufferedReader buf = new BufferedReader(isr);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *buf*. The resource is referenced in statements at the following lines: 150, 153, 160. The resource closure statement is at line: 163. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by BufferedReader.readLine. Either a) close *buf* in a try-finally block or b) close the resource by declaring *buf* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).